python/ray/air/execution/resources/resource_manager.py

```diff
@@ -0,0 +1,47 @@
+import abc
+
+from typing import List, Optional
+
+import ray
+from ray.air.execution.resources.request import (
+    ResourceRequest,
+    AllocatedResource,
+)
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class ResourceManager(abc.ABC):
+    def get_resource_futures(self) -> List[ray.ObjectRef]:
+        """Return futures for resources to await."""
+        return []
+
+    def request_resources(self, resource_request: ResourceRequest):
+        """Request resources, e.g. schedule placement group."""
+        raise NotImplementedError
+
+    def cancel_resource_request(self, resource_request: ResourceRequest):
+        """Cancel resource request, e.g. remove placement group."""
+        raise NotImplementedError
+
+    def has_resources_ready(self, resource_request: ResourceRequest) -> bool:
+        """Returns True if resources for the given request are available"""
+        raise NotImplementedError
+
+    def acquire_resources(
+        self, resource_request: ResourceRequest
+    ) -> Optional[AllocatedResource]:
+        """Acquire resources. Returns None if resources are not available."""
+        raise NotImplementedError
+
+    def free_resources(self, allocated_resources: AllocatedResource):
+        """Free resources from usage."""
+        raise NotImplementedError
+
+    def update_state(self):
```

Comment:
What exactly is this used for?

Maybe renaming to something like `notify_cluster_change` might make this more clear.

Reply:
Ah I guess this is explained in detail in the PlacementGroupResourceManager. I'll expand the docstring a bit